### ISSUE TITLE
“This should never happen”

### ISSUE THREAD

**Issue by Icelandjack** [ucsd-progsys/liquidhaskell#858]
```haskell
import Language.Haskell.Liquid.ProofCombinators

{-@ propPlusComm :: x:Int -> y:Int -> {x+y = y+x} @-}
propPlusComm :: Proof
propPlusComm = trivial *** QED
```

results in

```
$ liquid tkCd.hs
LiquidHaskell Copyright 2009-15 Regents of the University of California. All Rights Reserved.


**** DONE:  A-Normalization ****************************************************
 
liquid: <no location info>: Error: Uh oh.
    This should never happen! If you are seeing this message, 
please submit a bug report at 
https://github.com/ucsd-progsys/liquidhaskell/issues 
with this message and the source file that caused this error.

RefType.toType cannot handle: {VV : _ | x + y == y + x}
$
```

**Comment by nikivazou** [comment]
Correct! Terrible error message! Thanks for reporting! 

The fix though is to edit the Haskell implementation to 
```
propPlusComm :: Int -> Int -> Proof
propPlusComm _ _ = trivial *** QED
```

**Comment by ranjitjhala** [comment]
Yes, this *should* fail with the error message:

```
 /Users/rjhala/research/stack/liquidhaskell/tests/todo/T858.hs:3:21-60: Error: Specified Type Does Not Refine Haskell Type for Main.propPlusComm
 
 3 | {-@ propPlusComm :: x:Int -> y:Int -> {v:Proof | x+y = y+x} @-}
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The Liquid type
  
     GHC.Types.Int -> GHC.Types.Int -> ()
  
 is inconsistent with the Haskell type
  
     ()
```

@gridaphobe  this looks like another case of where the "hole resolution" is not, in fact, happening properly. @nikivazou is an easy fix to modify the parser to just treat `{p}` as `{v:Proof|p}` and *not* `{v:_ | p}` ?

**Comment by nikivazou** [comment]
@ranjitjhala we cannot modify the parser as you suggest, because `Proof` is not hard coded in the Liquid Haskell code, but only defined in the library. We just need to fix the hole resolution.

**Comment by gridaphobe** [comment]
fixed by #861
